[Ang Jing Heng, Marcus] iP

marcusjhang opened the pull request:
## **Muke**

> “Your mind is for having ideas, not holding them.” – David Allen ([source](#))

**Muke** helps you stay organized by keeping track of tasks so you can focus on what truly matters. It’s:

- User-friendly and text-based
- Simple and quick to master
- **Exceptionally FAST** for all your needs

### Get Started in Four Easy Steps:
1. [Download Muke](#https://github.com/marcusjhang/ip) from our official site.
2. Install by cloning onto your local device
3. Start adding your tasks.
4. Let Muke handle the rest for you! 😄

And the best part? **It’s completely FREE!**

### Key Features:
- ✅ Task Management
- ☐ Deadline Tracking (coming soon)
- ☐ Task Reminders (coming soon)

### Sample Java Code:
If you’re a Java enthusiast, Duke offers an excellent opportunity to refine your skills. Below is a snippet of the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args); // Launches the JavaFX application
    }
}
```

Feel free to explore and tweak the code to suit your learning journey. Duke is designed to be both functional and a learning tool for aspiring developers! 🚀

marcusjhang committed:
Level-0

marcusjhang committed:
level-1

marcusjhang committed:
Level-2

marcusjhang committed:
Level-3

marcusjhang committed:
Level-3

marcusjhang committed:
Level-4

marcusjhang committed:
A-TextUiTesting

marcusjhang committed:
refactor - task package created

marcusjhang committed:
Level-5

marcusjhang committed:
Level-6

marcusjhang committed:
A-Enums

marcusjhang committed:
added Storage edited Tasks edited Duke

marcusjhang committed:
edited notes.md

marcusjhang committed:
edited deadline and duke to accept datetime

marcusjhang committed:
added getter in Deadline, edited duke to show all tasks on a specific date

marcusjhang committed:
edited notes.md

marcusjhang committed:
edited Ui, Parser, TaskList packages

marcusjhang committed:
created duke package

marcusjhang committed:
added FindCommand, edited Parser, Ui to handle find command

marcusjhang committed:
complete show

marcusjhang committed:
print of list converted to getter in TaskList and show in Ui

marcusjhang committed:
changed print on date to getter and print on Ui

marcusjhang committed:
Merge branch 'branch-level-9'

marcusjhang committed:
edited .gitignore edited TaskList solved merge conflicts

marcusjhang committed:
initialized gradle

marcusjhang committed:
added tests for TaskList.java and Ui.java

marcusjhang committed:
added javadocs to all methods

marcusjhang committed:
edit sttings.json

marcusjhang committed:
Created Jar file, updated notes.md to give instructions on how to run Jar file

marcusjhang committed:
implemented fxml, built GUI using javafx

marcusjhang committed:
2nd commit for previous, didn't add these files

CYX22222003 commented on .vscode/java-formatter.xml:
It will be better to ignore this file as it seems to be specific to IDE.

CYX22222003 commented on .vscode/settings.json:
It will be better to ignore this file as it seems to be specific to IDE.

marcusjhang committed:
slightly refactored, yet to remove UI

CYX22222003 commented on src/main/java/duke/Duke.class:
It will be better to ignore this file.

CYX22222003 commented on .vscode/java-formatter.xml:
Should we ignore the configuration file that is specific to IDE?

CYX22222003 commented on src/main/java/duke/command/ListCommand.java:
Good to see you use OOP to abstract different types of Command

CYX22222003 commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,89 @@
+        String[] words = input.split(" ", 2);
+        String command = words[0];
+
+        switch (command) {
```
Good use of switch statement here

CYX22222003 commented on src/test/java/duke/ui/UiTest.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Test
+    public void testShowTaskAdded() {
```
Underscores may be used in test method names using the following three part format featureUnderTest_testScenario_expectedBehavior()

CYX22222003 commented on notes.md:
Very clear user manual

CYX22222003 reviewed (commented):
LGTM!
Very of good use of OOP and abstraction.
It will be better to remove binary files and IDE config files.

marcusjhang committed:
added asserts to Duke.java

kaikquah commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,92 @@
+        while (scanner.hasNextLine()) {
+            String line = scanner.nextLine();
+            String[] parts = line.split(" \\| ");
+            String taskType = parts[0];
```
good and clear naming throughout

kaikquah commented on src/main/java/duke/task/TaskType.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Represents the different types of tasks available in the Duke application.
+ */
+public enum TaskType {
```
nice use of enums here

kaikquah commented on src/test/java/duke/ui/UiTest.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @Test
+    public void testShowTaskRemoved() {
```
Could follow the required naming of test methods

kaikquah commented on src/main/java/duke/ui/Ui.java:
```diff
@@ -0,0 +1,177 @@
+    /**
+     * Displays the welcome message to the user.
+     */
+    public void showWelcome() {
```
good naming of methods throughout the whole class

kaikquah reviewed (commented):
LGTM overall. I'm very impressed with the code :). Just some nits about naming.

marcusjhang committed:
edit feature, remove dark mode

CYX22222003 commented on src/main/java/duke/command/MarkCommand.class:
Should this file be included in the repo?

CYX22222003 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,73 @@
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
+    protected TaskType taskType;
```
Is it really necessary to have a label of subtype? the printing of type information can be handled in the subclass.

marcusjhang committed:
add edit functionality

marcusjhang committed:
Add user guide to README.md

marcusjhang committed:
add Ui.png

marcusjhang committed:
edit README.md

marcusjhang committed:
Merge pull request #2 from marcusjhang/branch-A-UserGuide

edit README.md

marcusjhang committed:
edit duke to muke

There are several methods that use the word "Duke" or "duke".

This commit changes all instances of these wors to "Muke" or "duke" respectively.

marcusjhang committed:
edit README.md

marcusjhang committed:
edit README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
Update README.md

marcusjhang committed:
edit build file
